lib/src/lib.rs

```diff
@@ -999,21 +1000,41 @@ impl Repo {
     }
 
     fn chunk_writer(&self, rx: mpsc::Receiver<ChunkMessage>) {
+        let mut queue: VecDeque<(PathBuf, PathBuf, File)> = VecDeque::with_capacity(CHANNEL_SIZE);
+        // Cache paths to make sure we don't queue up an existing block
+        let mut queue_paths: HashSet<PathBuf> = HashSet::new();
+
+        fn flush(queue: &VecDeque<(PathBuf, PathBuf, File)>) {
+            for &(_, _, ref chunk_file) in queue.iter() {
+                chunk_file.sync_data().unwrap();
+                drop(chunk_file);
```

Comment:
I'm under impression that this is not dropping the `File` but only `&File`. The actual close on the file would be done on `queue.done()`. Even if I'm wrong, I think it would be more idiomatic and safer to use `queue.drain()` and not have to `queue.clear()`.